source/lac/scalapack.cc

```diff
@@ -263,6 +263,141 @@ ScaLAPACKMatrix<NumberType>::operator=(const FullMatrix<NumberType> &matrix)
 
 
 
+template <typename NumberType>
+void
+ScaLAPACKMatrix<NumberType>::copy_from(const LAPACKFullMatrix<NumberType> &B,
+                                       const unsigned int                  rank)
+{
+  if (n_rows * n_columns == 0)
+    return;
+
+  const unsigned int this_mpi_process(
+    Utilities::MPI::this_mpi_process(this->grid->mpi_communicator));
+
+#  ifdef DEBUG
+  Assert(Utilities::MPI::max(rank, this->grid->mpi_communicator) == rank,
+         ExcMessage("All processes have to call routine with identical rank"));
+  Assert(Utilities::MPI::min(rank, this->grid->mpi_communicator) == rank,
+         ExcMessage("All processes have to call routine with identical rank"));
+#  endif
+
+  // root process has to be active in the grid of A
+  if (this_mpi_process == rank)
+    {
+      Assert(grid->mpi_process_is_active, ExcInternalError());
+      Assert(n_rows == int(B.m()), ExcDimensionMismatch(n_rows, B.m()));
+      Assert(n_columns == int(B.n()), ExcDimensionMismatch(n_columns, B.n()));
+    }
+  // Create 1x1 grid for matrix B.
+  // The underlying grid for matrix B only contains the process #rank.
+  // This grid will be used to copy the serial matrix B to the distributed
+  // matrix using the ScaLAPACK routine pgemr2d.
+  MPI_Group group_A;
+  MPI_Comm_group(this->grid->mpi_communicator, &group_A);
+  const int              n = 1;
+  const std::vector<int> ranks(n, rank);
+  MPI_Group              group_B;
+  MPI_Group_incl(group_A, n, ranks.data(), &group_B);
+  MPI_Comm communicator_B;
+  MPI_Comm_create_group(MPI_COMM_WORLD, group_B, 0, &communicator_B);
```

Comment:
can we use `this->grid->mpi_communicator` here? We know that we copy to some rank within the communicator of this grid, right?